pkg/engine/virtualmachines.go

```diff
@@ -131,13 +131,16 @@ func CreateVirtualMachine(cs *api.ContainerService) VirtualMachineARM {
 	}
 
 	t, err := InitializeTemplateGenerator(Context{})
+	if err != nil {
+		panic(err)
```

Comment:
`panic()` is arguably as problematic as `log.Fatal`, but that was already the error-handling paradigm in this function.

Reply:
Could we change the function to return an error instead? I did a quick search and it doesn't seem like it's called in that many places so it wouldn't be that big of a change. Definitely agree we should try to avoid using panic().